### PR TITLE
Admin Style Changes

### PR DESCRIPTION
**Pull request by bislewl**
Changed admin styles to more closely resemble new colors of catalog (responsive_classic)

### CODE REVIEW

**Comment by zcwilt** [comment]
Hi 

While appreciate the work you have put in here, I'm seeing lots of ugliness :(
This is Ubuntu/Firefox

screenshot-v155 local 2016-01-23 16-58-40
screenshot-v155 local 2016-01-23 17-08-28

**Comment by bislewl** [comment]
I'll fix the login page. I suspect I may have taken it a bit too far with changing all those icons using font awesome. I'll revert that, as well. Did you want to close this pull request and then make changes, and re-request. Or modify and update this thread?

**Comment by zcwilt** [comment]
If you want to just update this, thats fine by me

**Comment by Zen4All** [comment]
Although I really appreciate all the work you put in, why use bootstrap?
Wasn't there a license compatibility problem? And Zen Cart 1.6.0 will be using foundation.

**Comment by barco57** [comment]
The license compatibility issue went away when Bootstrap switched too the MIT license.

**Comment by Zen4All** [comment]
@barco57 thanks

**Comment by bislewl** [comment]
@zcwilt I removed all the font awesome changes I did, and fix the admin login box styling issue. I also fixed the menus in the header. However I'm not sure if the 2nd menu should be the color that I have selected.

Thoughts?

**Comment by zcwilt** [comment]
@bislewl Hi Larry, will test again when I get a moment :)

Hopefully sometime tonight

**Comment by bislewl** [comment]
That's fine, at your leisure

Also rather than changing every admin page.
I added the JS to admin/includes/header.php
and the Bootstrap CSS to: admin/includes/stylesheet.css

It's not the "proper" way but it's functional

File: admin/includes/header_navigation.php [hunk 1]
```diff
@@ -11,6 +11,38 @@
...
+          <?php } ?>
+          <li class="upperMenuItems"><a href="<?php echo zen_href_link(FILENAME_DEFAULT, '', 'NONSSL'); ?>" class="headerLink"><?php echo HEADER_TITLE_TOP; ?></a></li>
+          <li class="upperMenuItems"><a href="<?php echo zen_catalog_href_link(FILENAME_DEFAULT); ?>" class="headerLink" target="_blank"><?php echo HEADER_TITLE_ONLINE_CATALOG; ?></a></li>
+          <li class="upperMenuItems"><a href="<?php echo zen_href_link(FILENAME_SERVER_INFO, '', 'NONSSL'); ?>" class="headerLink"><?php echo HEADER_TITLE_VERSION; ?></a></li>
+          <li class="upperMenuItems"><a href="<?php echo zen_href_link(FILENAME_ADMIN_ACCOUNT, '', 'NONSSL'); ?>" class="headerLink"><?php echo HEADER_TITLE_ACCOUNT; ?></a></li>
+          <li class="upperMenuItems"><a href="<?php echo zen_href_link(FILENAME_LOGOFF, '', 'NONSSL'); ?>" class="headerLink"><?php echo HEADER_TITLE_LOGOFF; ?></a></li>
+		</ul>
+	</div><!-- /.navbar-collapse -->
+</nav>
+
```
Comment (zcwilt) [inline admin/includes/header_navigation.php]: Had you meant to include a whole new navigation menu

**Comment by bislewl** [comment]
Yes.

Lines 37-41 as you have selected are only visible on smaller devices, and the upper ones are only available on larger devices, that way there is only 1 drop-down menu.

Are both showing for you?

**Comment by zcwilt** [comment]
yes, in a normal firefox browser I was seeing both menus

screenshot-v155 local 2016-01-28 22-42-52

**Comment by zcwilt** [comment]
Also 
screenshot-v155 local 2016-01-28 22-44-17

**Comment by bislewl** [comment]
@zcwilt  both of those should be resolved

**Comment by zcwilt** [comment]
k, so one problem related to removing the nde menu is that the nde javascript is still firing, as menu.js is included on all admin pages. This triggers a javascript error.

Now you could remove all references to menu.js but a quick fix is to edit menu.js and just under 
`  if (!document.getElementById)
  {
    return true;
  }
`

add 

`  if (!document.getElementById(menuid))
  {
    return true;
  }
`

**Comment by zcwilt** [comment]
@bislewl  forgot to mention that your fix did resolve my probs :)

**Comment by zcwilt** [comment]
also another minor messy

screenshot-v155 local 2016-01-29 17-02-16

**Comment by bislewl** [comment]
ok, I added the code you recommended also fixed the cancel button on a few other pages. Also Fixe the configuration Page Layout.

We really should look at somehow standardizing configuration.php/modules.php/record_artists.php maybe? They all have similar layouts

**Comment by zcwilt** [comment]
@bislewl yes that fixed it :)

> We really should look at somehow standardizing configuration.php/modules.php/record_artists.php maybe? They all have similar layouts

For simple CRUD pages we do now have the LEAD stuff in 1.6

**Comment by bislewl** [comment]
Any other changes needed?

**Comment by zcwilt** [comment]
Hi @bislewl 

Been through all the pages again and pretty sure these are my last 2 gripes :)

On configuration pages 
the edit box is detached.
configuration
 

And for other admin pages, we have lost some margin
adminpages

**Comment by zcwilt** [comment]
oops, didn't mean to close

**Comment by drbyte** [comment]
In addition to the margin issues @zcwilt mentioned, the Change Password section needs some alignment fixing.

screen shot 2016-02-12 at 6 10 30 pm

**Comment by drbyte** [comment]
Another margin issue is in the category/products area ... the left-most "ID" column seems to have lost the bit of padding between itself and the next column.

screen shot 2016-02-12 at 6 22 34 pm

**Comment by drbyte** [comment]
The submit buttons on the "Admin Profiles 'edit'" page are misaligned.

screen shot 2016-02-12 at 7 02 12 pm

**Comment by drbyte** [comment]
Feedback on the overall changes:
a) On desktop the very tall padding of the actual menu items feels like wasted space and unnecessary.  On a mobile device it makes some sense for touchability.
At the very least, perhaps the font size could be increased a bit?

b) I'm itching for more color ... mainly because the lack of distinction that there's actually a menu there. The `.navbar-default` is currently `#f8f8f8`. Maybe something like `#d7d6cc` would create more distinction?

I've been experimenting with the following, added to menu.css, but only tested it on desktop so far:

```css
.navbar-default {
  background-color: #d7d6cc;
}
.navbar-nav li>a.dropdown-toggle {
  font-size: 1.3em;
  color: #333;
}
@media (min-width: 768px) {
  .navbar-nav>li>a {
    padding-top: 10px;
    padding-bottom: 10px;
  }
  .navbar {
    min-height: 30px;
  }
}
```

**Comment by drbyte** [comment]
Hi @bislewl ... anything we can do to help?

**Comment by bislewl** [comment]
@drbyte I have made the changes as you request I also modified the header as well, to the check for updates as well. Something to consider in the future is to standardize the buttons. Other question on a semi side not. Why are the login/forgot password/change password not using zen_draw_form() ?

**Comment by drbyte** [comment]
> Something to consider in the future is to standardize the buttons.

Agreed. Normally I wouldn't do this sort of overhaul in a point release. I hope we can do those kinds of improvements in the v160 code though. Lots of ripple effects to consider there.

**Comment by drbyte** [comment]
> Other question on a semi side not. Why are the login/forgot password/change password not using zen_draw_form() ?

I don't know if there's a technical reason for that, or if it was just never standardized. Haven't tested changing it.

**Comment by drbyte** [comment]
> I have made the changes as you request

Thanks. I'll set up another site with it and do some more clicking around :)

**Comment by drbyte** [comment]
NOTE: There are some merge conflicts because your branch is based on work done awhile back, and haven't git-merged the changes we've made in the meantime to some of the files you've touched.

How comfortable are you with resolving those conflicts? (ie; use git locally to do a pull from the master v155 branch into this branch, and then resolve all conflicts, complete the merge, and push the result back up to github)

**Comment by bislewl** [comment]
> How comfortable are you with resolving those conflicts? (ie; use git locally to do a pull from the master v155 branch into this branch, and then resolve all conflicts, complete the merge, and push the result back up to github)

I'll give it a try, what's the worst that could happen right?

**Comment by bislewl** [comment]
one moment i'm fixing a small bug

**Comment by bislewl** [comment]
All those same lines what are your thoughts on add a label draw function to the functions/html_output.php for
something like:
`function zen_draw_label($text,$for){
  $label = '<label for="'.$for.'">'.$text.'</label>';
  return $label;
`}``

**Comment by drbyte** [comment]
referring to: 

> admin/includes/languages/english.php
>
> @212
>  define('BOX_HEADING_ADMIN_ACCESS', 'Admin Access Management');

**Comment by zcwilt** [comment]
:+1:

**Comment by drbyte** [comment]
:+1: 

I'm satisfied that this PR does what it set out to do.
And I don't see any glaring issues.

There will be some challenges for people upgrading their plugins where those plugins add small alterations to the files affected here.  But I don't think there are any backward-compatibility "breaks" caused by this (ie: changes to functions/etc where old code would work differently).

**Comment by barco57** [comment]
Ok, I asked this in private and didn't get an answer, so I will ask in public....why, when 1.6.0 is using Zurb Foundation in the admin is 1.5.5 using Bootstrap in the admin?

**Comment by Zen4All** [comment]
I was wondering the same thing :wink: 

> https://github.com/zencart/zc-v1-series/pull/793#issuecomment-174216537

**Comment by drbyte** [comment]
> @Zen4All commented on Jan 23
> Although I really appreciate all the work you put in, why use bootstrap?
> Wasn't there a license compatibility problem? And Zen Cart 1.6.0 will be using foundation.

> @barco57 commented on Jan 23
> The license compatibility issue went away when Bootstrap switched too the MIT license.
 
> @Zen4All commented on Jan 23
> @barco57 thanks

> @barco57 commented on Feb 22
> Ok, I asked this in private and didn't get an answer, so I will ask in public....why, when 1.6.0 is using Zurb Foundation in the admin is 1.5.5 using Bootstrap in the admin?

> @Zen4All commented on Feb 22
> I was wondering the same thing 


Are you saying you're offering to convert it to Foundation?

**Comment by bislewl** [comment]
Personally I would rather use bootstrap in 1.6.0, but that is just my 2 cents, and the development is already done thou. I think bootstrap is easier to understand, and has more support. Unfortunately we have come to a "fork in the road". What is everyone's opinion of bootstrap or foundation?

**Comment by Zen4All** [comment]
To me it just seemed like doing the job twice. (Sadly) my knowledge of both bootstrap and foundation is limited, because I do not see/know the advantages of these frameworks

**Comment by barco57** [comment]
I have been considering my answer for a couple of days...

I don't think the opinion of one framework over the other is important unless one version or the other is going to be rewritten to be consistent with the other. If it doesn't matter that bootstrap is used in one and foundation in the other, then by all means carry on, but if it does matter, let the choosing begin.

Being Foundation has been in the v1.6 admin since Jul 17, 2013 and work that was done by me (minor) and others has been done with foundation, I would vote for Foundation, only for the fact it makes all of that work worthless to some degree. Being my part in that was a small contribution, reworking it would not be a problem if it were to change to Bootstrap but others with more contributions much more work too redo should speak up as too their opinion.

The problem as I see it is that for 1.5.5 there will be addon developers that will happily go about updating their addons, and gladly use bootstrap classes in their code for display....then v1.6 releases with Foundation (as it stands currently) and not only do they have to update their addons functionality in regard to how it works, but redo the display code for it...and being there will still be people using 1.5.5 for a while, possibly have to maintain to versions as they bug fix things over time.

If I am wrong, somebody please point it out, I'll be happy to say "I'm stupid and didn't understand"

**Comment by barco57** [comment]
I do wish I had brought this up when I originally answered the question about Bootstrap licensing history but I wasn't thinking about the bigger picture at the time.

**Comment by bislewl** [comment]
Several of the TM themes use Bootstrap, that's why I was leaning that way.

I guess my question is....regardless of the work that is done.

What is better for ZenCart? Bootstrap or Foundation? Which will provide a better future? Which will be better for those involved? Is there any way to take a poll?

**Comment by DivaVocals** [comment]
Personally I don't think which framework is better is a worthwhile or productive discussion.. 6 of 1, half dozen of the other.. It will be a long, protracted, opinion laced discussion with no real resolution. I think the REAL question to consider is the one asked by @Zen4All & @barco57. Why would v1.5.5 use a different framework than v1.6.0? I personally don't care one way or another.. I think there are just as many pros and cons to BOTH frameworks, and ultimately it doesn't matter. Pick one and let's be consistent.. Bootstrap appears to be the most popular one of the two frameworks. However, if work has already been done for v1.6.0 with the Foundation frmework, then the question really is not which framework is better, but does it make sense to continue down the path of introducing a different framework into v1.5.5's admin?

**Comment by drbyte** [comment]
Hi all,

Thanks for the feedback and candid discussion.  We're looking at this seriously ... and leaning heavily towards embracing Bootstrap.

Kinda don't want to thrust users into something that's "obviously a Bootstrap site, cuz it's got that cookie-cutter look" though.

**Comment by jtwebsitedesign** [comment]
I absolutely agree that either framework is perfectly fine, and as others have said here, it's best to keep it consistent for each release.

Some food for thought:
How much work has been performed (on core & plugins) using Bootstrap in v1.5.5 so far?
And the same question for Foundation in v1.6.0?
Will converting v1.5.5 to Foundation delay the release?
Will converting v1.6.0 to Bootstrap delay the release?
Should we introduce the "new look" in v1.6.0 instead of v1.5.5?
If so, should we include either framework in v1.5.5 at all?

In my opinion, a new framework/design would be more welcomed on the major release, and it would give us time to make a solid decision one way or the other. Then we're one step closer to releasing v1.5.5 as a successor to v1.5.4, and we can focus on redesign in the next gen release.

I like bislewl's idea about a poll:
"What is better for ZenCart? Bootstrap or Foundation? Which will provide a better future? Which will be better for those involved?"

My vote is for either/or. Bootstrap is more common, so there is more documentation/community surrounding it. Foundation functions about the same, but as drbyte mentions, it has more design flexibility. There is also something to be said for the work already put into v1.6.0. I'm not too invested one way or the other, but I'm sure others would like to weigh in.

**Comment by bislewl** [comment]
@drbyte is there a way on Zen-Cart.com to do a poll?

**Comment by drbyte** [comment]
Yes, but I'd rather people write actual comments, instead of just cast a vote.

Do you really think a poll will actually give something useful? My gut tells me it'll  we'll get one of two outcomes: Either a 45/45/10 split with equal parts saying one framework or the other and a small portion saying neither.  Or it'll be 75/25 with majority being BS and rest being other/neither.

I'd rather know their "why", not just their preference.

**Comment by drbyte** [comment]
In the interest of asking the "why", I've started a thread: https://www.zen-cart.com/showthread.php?219608-Bootstrap-or-Foundation-And-Why

**Comment by jtwebsitedesign** [comment]
Agreed, a poll based on discussion would be more productive. And I personally think the new framework should be implemented in v1.6.0, unless there is a particular reason we need it in v1.5.5.

**Comment by drbyte** [comment]
Another opinion question: 

For the CSS/JS framework that's in the Admin, should it be served via CDN, or served local?

Admin performance "caching" isn't as big a deal as catalog-side, but there *is* the issue of doing dev work offline, such as on a laptop on the train without wifi, etc.

ie: how often do you do dev work offline, disconnected from the internet and would need the assets served locally

**Comment by DivaVocals** [comment]
Personally speaking I often work offline when I'm working on my laptop from my day job where our network's security policy prevents me from using the internal wifi (internal wifi is for company devices and guest wifi is for guests onsite doing business with the company only). So CDN with local fallback perhaps??

**Comment by DivaVocals** [comment]
>**My vote is for either/or. Bootstrap is more common, so there is more documentation/community surrounding it.** Foundation functions about the same, but as drbyte mentions, it has more design flexibility. **There is also something to be said for the work already put into v1.6.0. I'm not too invested one way or the other**, but I'm sure others would like to weigh in.

I agree with this statement..

**Comment by Zen4All** [comment]
I do a lot of work when traveling, and very often there is no internet connection, or it is expensive. So my vote goes to CDN with local fallback to.

**Comment by jtwebsitedesign** [comment]
What are the security implications of serving from a CDN? Will more PCI scanning need to be implemented as well?

**Pull request merged**